libs/langchain-google-common/src/utils/gemini.ts

```diff
@@ -78,27 +87,70 @@ export function messageContentToParts(content: MessageContent): GeminiPart[] {
       : content;
 
   // eslint-disable-next-line array-callback-return
-  const parts: GeminiPart[] = messageContent.map((content) => {
-    // eslint-disable-next-line default-case
-    switch (content.type) {
-      case "text":
-        return messageContentText(content);
-      case "image_url":
-        return messageContentImageUrl(content);
+  const parts: GeminiPart[] = messageContent
+    .map((content) => {
+      // eslint-disable-next-line default-case
```

Comment:
I think we want to throw an error if an unsupported type comes through here, instead of silently failing.

Reply:
```typescript
default:
  throw new Error(`Unsupported type received while converting message to message parts: ${content.type}`)
```

Reply:
At the moment, this can only have these two values.

I'm uncomfortable with the notion that a future update that does introduce a new content type could break this code and prevent it from working until we get a fix in.

But I see that's how it's handled elsewhere... so I guess I have no strong objection.